config/src/main/java/com/typesafe/config/impl/ConfigDocumentParser.java

```diff
@@ -410,9 +425,9 @@ private ConfigNodeComplexValue parseObject(boolean hadOpenCurly) {
                     putBack(t);
                     break;
                 } else if (flavor != ConfigSyntax.JSON && isIncludeKeyword(t)) {
-                    objectNodes.add(new ConfigNodeSingleToken(t));
-                    parseInclude(objectNodes);
-
+                    ArrayList<AbstractConfigNode> includeNodes = new ArrayList<AbstractConfigNode>();
+                    includeNodes.add(new ConfigNodeSingleToken(t));
+                    objectNodes.add(parseInclude(includeNodes));
```

Comment:
if the tests are passing I must be confused here, but it sure looks like only the token `include` gets passed to parseInclude, and then parseInclude is parsing the filename after `include` as well (which I don't see how it has).
I'm sure I'll figure it out after I post this comment ;-)